odoo/addons/base/models/assetsbundle.py

```diff
@@ -246,43 +264,66 @@ def clean_attachments(self, type):
 
         return True
 
-    def get_attachments(self, type, ignore_version=False):
+    def get_attachments(self, extension, version_mode=''):
         """ Return the ir.attachment records for a given bundle. This method takes care of mitigating
         an issue happening when parallel transactions generate the same bundle: while the file is not
         duplicated on the filestore (as it is stored according to its hash), there are multiple
         ir.attachment records referencing the same version of a bundle. As we don't want to source
         multiple time the same bundle in our `to_html` function, we group our ir.attachment records
         by file name and only return the one with the max id for each group.
+        :param extension: file extension (js, min.js, css)
+        :param version_mode: if 'IGNORE', the version does not matter (.../%-%/...),
+                             elif 'NONE', don't have a version in its url (.../%/...),
+                             elif '', has the version in its url (.../%-version/...).
         """
-        unique = "%" if ignore_version else self.version
-        url_pattern = self.get_asset_url(
+        assert version_mode in ('', 'IGNORE', 'NONE')
+        unique = "%"
+        if version_mode == 'IGNORE':
+            unique = "%"
+        elif version_mode == 'NONE':
+            unique = ""
+        else:
+            unique = self.version
+
+        url_pattern_with_version = self.get_asset_url(
             unique=unique,
-            extra='%s' % ('rtl/' if type == 'css' and self.user_direction == 'rtl' else ''),
+            extra='%s' % ('rtl/' if extension == 'css' and self.user_direction == 'rtl' else ''),
             name=self.name,
             sep='',
-            type='.%s' % type
+            type='.%s' % extension
         )
         self.env.cr.execute("""
-             SELECT max(id)
-               FROM ir_attachment
-              WHERE create_uid = %s
+            SELECT max(id)
+            FROM ir_attachment
+            WHERE create_uid = %s
                 AND url like %s
-           GROUP BY name
-           ORDER BY name
-         """, [SUPERUSER_ID, url_pattern])
+        GROUP BY name
+        ORDER BY name
+        """, [SUPERUSER_ID, url_pattern_with_version])
```

Comment:
why mess up the formatting of the query without good reason? it's column-aligned after the keywords, which is one of the popular ways to format SQL queries ;)